### PR TITLE
 Add data inputs example for compass

### PR DESCRIPTION
**Pull request by VitaminPSG**
The UXF SDK exposes [DataInputsManager](https://docs.mapbox.com/android/navigation/api/uxframework/1.0.0-beta.37/data-inputs-api/com.mapbox.dash.sdk.data.inputs.api/-dash-data-inputs-manager/?query=interface%20DashDataInputsManager) interface to allow a client to send sensor data coming from a device (or vehicle) to the UXF navigation engine.

Today, this data is supported by the mentioned API:

* **altimeterData**
* **accelerometerData**
* **rawGyroscopeData**
* **compassData**
* **etcGateInfo**
* **motionData**
* **odometryData**
* **rawGnssData**

In order to have access to DataInputsManager you should:

1.  Add a corresponding module of UXF SDK in your dependencies:

`implementation("com.mapbox.navigationux:data-inputs:1.0.0-beta.37.1")`

2. Enable this functionality in DSL API:

```
 dataInputs {
     enabled = true
 }
```
3. That's it. Now, you can use Data Inputs API via Dash.controller like this:

```
sampleSensorEventManager?.compassData?.observeWhenStarted(this) {
    if (enabled) {
        Log.d(TAG, "Updating compass data: $it")
        Dash.controller.updateCompassData(it)
    }
}
```

This PR shows how compass data from an android device can be passed to data input manager.

### CODE REVIEW

**Comment by alexzatsepin** [comment]
@Zayankovsky ptal.

File: app/src/main/java/com/mapbox/dash/example/MainActivity.kt [hunk 1]
```diff
@@ -444,6 +463,18 @@ class MainActivity : DrawerActivity() {
...
+            switch = menuBinding.customCompassDataInputs,
+            state = setCustomCompassDataInputs,
+        ) { enabled ->
+            sampleSensorEventManager?.compassData?.observeWhenStarted(this) {
+                if (enabled) {
+                    Log.d(TAG, "Updating compass data: $it")
+                    Dash.controller.updateCompassData(it)
+                }
+            }
+        }
```
Comment (Zayankovsky) [inline app/src/main/java/com/mapbox/dash/example/MainActivity.kt:476]: This launches a new coroutine every time a switch is checked or unchecked. Should be fixed, e.g. like this:
```suggestion
        menuBinding.customCompassDataInputs.setOnCheckedChangeListener { _, isChecked ->
            setCustomCompassDataInputs.value = isChecked
        }

        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED) {
                setCustomCompassDataInputs.collectLatest { enabled ->
                    if (!enabled) return@collectLatest
                    sampleSensorEventManager?.compassData?.collect {
                        Log.d(TAG, "Updating compass data: $it")
                        Dash.controller.updateCompassData(it)
                    }
                }
            }
        }
```

File: app/src/main/java/com/mapbox/dash/example/MainActivity.kt [hunk 2]
```diff
@@ -444,6 +463,18 @@ class MainActivity : DrawerActivity() {
...
         }
 
+        bindSwitch(
+            switch = menuBinding.customCompassDataInputs,
+            state = setCustomCompassDataInputs,
+        ) { enabled ->
+            sampleSensorEventManager?.compassData?.observeWhenStarted(this) {
+                if (enabled) {
+                    Log.d(TAG, "Updating compass data: $it")
+                    Dash.controller.updateCompassData(it)
```
Comment (Zayankovsky) [inline app/src/main/java/com/mapbox/dash/example/MainActivity.kt:473]: Is there a way to reset the data when the switch is unchecked?

Comment (VitaminPSG) [reply app/src/main/java/com/mapbox/dash/example/MainActivity.kt:448]: No

File: app/src/main/java/com/mapbox/dash/example/MainActivity.kt [hunk 3]
```diff
@@ -419,6 +443,34 @@ class MainActivity : DrawerActivity() {
...
+                }
+            }
+        }
+
```
Comment (Zayankovsky) [inline app/src/main/java/com/mapbox/dash/example/MainActivity.kt:457]:
```suggestion
```

**Review by Zayankovsky — APPROVED** [review]

**Comment by alexzatsepin** [comment]
@VitaminPSG let's merge this PR once you have a chance.

**Pull request merged**